### PR TITLE
fix(client): Only make profile requests if a valid accessToken exists.

### PR DESCRIPTION
**Pull request by shane-tomlinson**

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@zaach, @vladikoff, @jrgm, @seanmonstar - I'm PTO today, and I am unsure if this PR is against the current 42.1 sha. I think this is a minimal fix for the problem @seanmonstar notes in [1].

ref #2832. I think for the train 44, we need to take a good look at this code. I saw some other anomalies today (double cert signing, requesting profile information on signout) that made me say "eh?"

[1] - https://bugzilla.mozilla.org/show_bug.cgi?id=1191052#c3

File: app/scripts/models/account.js [hunk 1]
```diff
@@ -350,15 +350,24 @@ define([
           .then(function (client) {
             profileClient = client;
             var accessToken = self.get('accessToken');
-            return profileClient[method].apply(profileClient, [accessToken].concat(args));
+            if (accessToken) {
+              return profileClient[method].apply(profileClient, [accessToken].concat(args));
+            } else {
+              throw ProfileClient.Errors.toError('UNAUTHORIZED');
```
Comment (vladikoff) [inline app/scripts/models/account.js]: I don't think it should `throw UNAUTHORIZED`, let me re-PR something ...

Comment (vladikoff) [inline app/scripts/models/account.js]: Actually that should work, testing more

**Comment by vladikoff** [comment]
Thanks Shane!

File: app/scripts/models/account.js (hunk 1 above)
Comment (zaach) [inline app/scripts/models/account.js]: Hmm, the token fetch promise should not resolve if fetching the token fails... Will look more later.

File: app/tests/spec/models/account.js [hunk 2]
```diff
@@ -478,6 +478,38 @@ function (chai, sinon, p, Constants, Assertion, ProfileClient,
       });
     });
 
+    describe('without a valid sessionToken, without an access token', function () {
+      beforeEach(function () {
+        account.unset('sessionToken', 'expired session token');
```
Comment (zaach) [inline app/tests/spec/models/account.js]: is this supposed to be setting or unsetting the value?

Comment (shane-tomlinson) [inline app/tests/spec/models/account.js]: doh, it's supposed to set an expired sessionToken. Same end effect though.

**Pull request merged**